openmc/model/triso.py

```diff
@@ -153,3 +162,843 @@ def create_triso_lattice(trisos, lower_left, pitch, shape, background):
+            Indices of lattice cells.
+
+        """
+
+        if cl is None:
+            cl = cell_length
+
+        r = [[a/cl[i] for a in [p[i]-d, p[i], p[i]+d] if a > 0 and
+             a < domain_length] for i in range(3)]
+
+        return list(itertools.product(*({int(i) for i in j} for j in r)))
+
+
+    def cell_list_cylinder(p, d, cl=None):
+        """Return the indices of all cells within the given distance of the
+        point.
+
+        Parameters
+        ----------
+        p : list of float
+            Cartesian coordinates of particle center.
+        d : float
+            Find all lattice cells that are within a radius of length 'd' of
+            the particle center.
+        cl : list of float
+            Length of the lattice cells in x-, y-, and z-directions.
+
+        Returns
+        -------
+        list of tuple of int
+            Indices of lattice cells.
+
+        """
+
+        if cl is None:
+            cl = cell_length
+
+        x,y = [[(a + domain_radius)/cl[i] for a in [p[i]-d, p[i], p[i]+d]
+               if a > -domain_radius and a < domain_radius] for i in range(2)]
+
+        z = [a/cl[2] for a in [p[2]-d, p[2], p[2]+d] if a > 0
+             and a < domain_length]
+
+        return list(itertools.product(*({int(i) for i in j} for j in (x,y,z))))
+
+
+    def cell_list_sphere(p, d, cl=None):
+        """Return the indices of all cells within the given distance of the
+        point.
+
+        Parameters
+        ----------
+        p : list of float
+            Cartesian coordinates of particle center.
+        d : float
+            Find all lattice cells that are within a radius of length 'd' of
+            the particle center.
+        cl : list of float
+            Length of the lattice cells in x-, y-, and z-directions.
+
+        Returns
+        -------
+        list of tuple of int
+            Indices of lattice cells.
+
+        """
+
+        if cl is None:
+            cl = cell_length
+
+        r = [[(a + domain_radius)/cl[i] for a in [p[i]-d, p[i], p[i]+d]
+             if a > -domain_radius and a < domain_radius] for i in range(3)]
+
+        return list(itertools.product(*({int(i) for i in j} for j in r)))
+
+
+    def random_sequential_pack():
+        """Random sequential packing of particles whose radius is determined by
+        initial packing fraction.
+
+        Returns
+        ------
+        numpy.ndarray
+            Cartesian coordinates of centers of TRISO particles.
+
+        """
+
+        # Set parameters for initial random sequential packing of particles.
+        r = (3/4*initial_packing_fraction*domain_volume/(pi*n_particles))**(1/3)
+        d = 2*r
+        sqd = d**2
+        cl = get_cell_length(r)
+
+        particles = []
+        mesh = defaultdict(list)
+
+        for i in range(n_particles):
+            # Randomly sample new center coordinates while there are any overlaps
+            while True:
+               p = random_point()
```

Comment:
Inconsistent indentation here (3 spaces). This whole block needs to move forward one space.